### PR TITLE
fix(2325): fixed the issue for default CGS to settle deferred transfers

### PR DESCRIPTION
**Pull request by vgenev**

### CODE REVIEW

File: src/models/transferSettlement/facade.js [hunk 1]
```diff
@@ -333,8 +334,25 @@ async function getSettlementModelByTransferId (transferId, settlementGranularity
...
+    switch (settlementGranularityName) {
+      case SettlementEnum.settlementGranularityName.GROSS: {
+        const netModelWithCurrency = allSettlementModels.filter(sm => (sm.currencyId === transferCurrency && sm.settlementGranularityId === SettlementEnum.SettlementGranularity.NET))
+        if (netModelWithCurrency.length === 0) {
+          const defaultGrossSettlementModel = allSettlementModels.filter(sm => (sm.currencyId === null && sm.settlementGranularityId === SettlementEnum.SettlementGranularity.GROSS))
+          return defaultGrossSettlementModel
+        }
+        break
+      }
+      case SettlementEnum.settlementGranularityName.NET: {
```
Comment (mdebarros) [inline src/models/transferSettlement/facade.js:347]: Is there any reason that you need to check for GROSS within the Net case?

I think you only need to do this:

```suggestion
      case SettlementEnum.settlementGranularityName.NET: {
        return allSettlementModels.filter(sm => (sm.currencyId === transferCurrency && sm.settlementGranularityId === SettlementEnum.SettlementGranularity.NET))
      }
```

Comment (mdebarros) [inline src/models/transferSettlement/facade.js:347]: Is there any reason that you need to check for NET within the Gross case?

I think you only need to do this:

```suggestion
      case SettlementEnum.settlementGranularityName.GROSS: {
        return allSettlementModels.filter(sm => (sm.currencyId === transferCurrency && sm.settlementGranularityId === SettlementEnum.SettlementGranularity.GROSS))
      }
```

**Review by mdebarros — CHANGES_REQUESTED** [review]
See my comments.

File: src/models/transferSettlement/facade.js (hunk 1 above)
Comment (vgenev) [reply src/models/transferSettlement/facade.js:347]: yes, to not look for default, in case of the transfer is in currency that has the opposite model.

Comment (vgenev) [reply src/models/transferSettlement/facade.js:347]: yes, to not look for default, in case of the transfer is in currency that has the opposite model.

**Review by mdebarros — APPROVED** [review]
+1

**Pull request merged**